packages/client/src/pages/MyTrecipes/MyTrecipes.tsx

```diff
@@ -1,60 +1,221 @@
 import React from 'react';
 import './MyTrecipes.scss';
-import { FilterButton } from './Filter/FilterButton';
+import { FilterButton, FilterButtonTypes } from './Filter/FilterButton';
 import { Button } from '../../components/Button/Button';
 import TrecipeCard from './TrecipeCard/TrecipeCard';
-import { FilterSelector } from './Filter/FilterSelector';
+import { FilterSelector, FilterSelectorTypes } from './Filter/FilterSelector';
 import { IconProp } from '@fortawesome/fontawesome-svg-core';
 import { connect } from 'react-redux';
 import { RootState } from '../../redux';
 import { fetchAllTrecipes } from '../../redux/TrecipeList/action';
 import { bindActionCreators, Dispatch } from 'redux';
 import { showModal } from '../../redux/Modal/action';
-import { Link, withRouter, RouteComponentProps } from 'react-router-dom';
+import { Link, RouteComponentProps, withRouter } from 'react-router-dom';
 import TrecipePopup, { TrecipePopupType } from '../../components/TrecipePopup/TrecipePopup';
 import Trecipe from '../../../../shared/models/trecipe';
 
 type MyTrecipesProps = ReturnType<typeof mapStateToProps> &
     ReturnType<typeof mapDispatchToProps> &
     RouteComponentProps;
 
-class MyTrecipes extends React.Component<MyTrecipesProps, {}> {
+interface FilterWithState {
+    filterName: string;
+    selected: boolean;
+}
+
+export interface MyTrecipesState {
+    filterStates: Array<FilterWithState>;
+    privacyState: FilterSelectorTypes;
+    filteredTrecipesWithProgress: Array<Trecipe>;
+    filteredTrecipesWithPrivacy: Array<Trecipe>;
```

Comment:
Personally I won't store these in state, because whenever we update the state, React will re-render the component. Since we update the state multiple times on each filter change, the component is being unnecessary re-rendered multiple times. 
If you are busy, I guess we can log an issue and maybe fix this later 😛

Reply:
hacky hacky, let me fix them